### PR TITLE
adios2: Bump version to v2.8.0 and cleanup package code

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

**Comment by ax3l** [comment]
I see this in CI with regards to the `ecp-data-vis-sdk` package:
```
==> Starting concretization
==> Error: ecp-data-vis-sdk+adios2+ascent+cinema+darshan+faodel+hdf5+paraview+pnetcdf+sz+unifyfs+veloc+vtkm+zfp is unsatisfiable, errors are:
  A conflict was triggered
    To see full clingo unsat cores, re-run with `spack --show-cores=full`
    For full, subset-minimal unsat cores, re-run with `spack --show-cores=minimized
    Warning: This may take (up to) hours for some specs
```

File: var/spack/repos/builtin/packages/adios2/package.py [hunk 1]
```diff
@@ -81,7 +83,7 @@ class Adios2(CMakePackage):
     conflicts('+shared ~pic')
 
     # DataMan needs dlopen
-    conflicts('+dataman', when='~shared')
+    conflicts('+dataman', when='+shared')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/adios2/package.py]: Changed by mistake? (see comment)
```suggestion
    conflicts('+dataman', when='~shared')
```

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/adios2/package.py]: I'm not even sure why that's there. There's no inherent issue with using the dataman engine with either static or shared.  I'm digging into it and the sdk failure

File: var/spack/repos/builtin/packages/adios2/package.py [hunk 2]
```diff
@@ -34,22 +36,22 @@ class Adios2(CMakePackage):
             description='CMake build type',
             values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
     variant('shared', default=True,
-            description='Also build shared libraries')
-    variant('pic', default=True,
-            description='Enable position independent code '
-                        '(for usage of static in shared downstream deps)')
+            description='Build shared libraries')
+    variant('pic', default=False,
```
Comment (MichaelLaufer) [inline var/spack/repos/builtin/packages/adios2/package.py]: FYI, this change (default pic=False) caused build errors for me. Requesting +pic in the spec solved it for me. Any reason why the default was changed? Thanks

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/adios2/package.py]: It was an unintentional change, it should default to True

**Comment by chuckatkins** [comment]
This should be good to go now once ci finishes

**Review by kwryankrattiger — APPROVED** [review]

**Comment by chuckatkins** [comment]
@ax3l ?

**Review by ax3l — APPROVED** [review]
Thanks a lot @chuckatkins :tada:

**Pull request merged**